base_layer/core/src/proof_of_work/monero_rx/merkle_tree.rs

```diff
@@ -132,6 +131,38 @@ pub struct MerkleProof {
     path_bitmap: u32,
 }
 
+impl BorshSerialize for MerkleProof {
+    fn serialize<W: Write>(&self, writer: &mut W) -> io::Result<()> {
+        BorshSerialize::serialize(
+            &(u32::try_from(self.branch.len()).map_err(|_| ErrorKind::InvalidInput)?).to_le_bytes(),
```

Comment:
Borsh should be able to serialize a u32 without first converting it to bytes
```suggestion
            &u32::try_from(self.branch.len()).map_err(|_| ErrorKind::InvalidInput)?
```